src/TestFramework/Coverage/CodeCoverageData.php

```diff
@@ -35,13 +35,12 @@
 
 namespace Infection\TestFramework\Coverage;
 
-use Infection\MutationInterface;
 use Infection\TestFramework\PhpUnit\Coverage\CoverageXmlParser;
 
 /**
  * @internal
  */
-class CodeCoverageData
+final class CodeCoverageData implements CodeCoverageDataInterface
```

Comment:
it's not really clear for me:

- why switching to an interface + class when there is only a single implementation?
- I quite dislike `CodeCoverageData` as the name, and this IMO is the issue with naming things with the `Interface` prefix.

If you have `FooInterface`, it's easy to name your implementation `Foo` which is not helpful as it doesn't explain what `Foo` is about at all. On the other hand, if you have `Foo` as the interface, besides being more clear as a consumer (because as a consumer you don't care if it's an interface or not), it also force you to choose a better suited name for its implementations.

In this peculiar case, I think `LineCodeCoverage` or `XmlLineCodeCoverage` would be a better suited name?

Reply:
`LineCodeCoverage` doesn't fit the purpose of this class in my opinion, as the class is also used to check if a file has any coverage at all.

However, for the interface it may be called `CodeCoverageData`, while the implementation is `XMLCodeCoverageData`, as it is a representatation of the xml output from phpunit.

Reply:
>LineCodeCoverage doesn't fit the purpose of this class in my opinion, as the class is also used to check if a file has any coverage at all.

I don't really see the problem here: it uses an XML based line code coverage report to provide all this information. Also from an API PoV it makes sense that if you can provide the tests for a given mutation, you can tell if there is any test for the given file at all